styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -209,6 +225,22 @@ private void initialize(WorkflowInstance workflowInstance) {
     }
   }
 
+  private void verifyCounter(Event event, long expectedCounter, RunState currentRunState) {
+    final long currentCounter = currentRunState.counter();
+    if (currentCounter > expectedCounter) {
+      final String message = "Stale event encountered. Expected counter is "
+                             + expectedCounter  + " but current counter is "
+                             + currentCounter + ". Discarding event " + event;
+      LOG.warn(message);
+      throw new StaleEventException(message);
```

Comment:
I think we can either ignore this even completely or log it as `debug`. We can just throw `RuntimeException` instead of having a new exception type.

Reply:
As we talked irl, we can keep the `StaleEventException` but I changed the log level to `debug`